npm/pkg/dataplane/dataplane_windows.go

```diff
@@ -43,6 +47,34 @@ func (dp *DataPlane) initializeDataPlane() error {
 	return nil
 }
 
+func (dp *DataPlane) getNetworkInfo() error {
+	err := dp.setNetworkIDByName(AzureNetworkName)
+	if err == nil {
+		return nil
+	}
+
+	if isNetworkNotFoundErr(err) {
+		retryNumber := 0
+		ticker := time.NewTicker(time.Second * time.Duration(maxNoNetSleepTime))
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				err := dp.setNetworkIDByName(AzureNetworkName)
+				if err == nil || !isNetworkNotFoundErr(err) {
```

Comment:
Discrepancy between code in this if-statment and outside. For an error that isn't networkNotFound, we return nil here, but err outside